[Watchdog Timer] Clear timer for already terminated process

gag1jain opened the pull request:
Summary:
handling cases where worker process is terminated w/o releasing the timer request, this scenario causes reaping of process at expiry.

removing the non-existent process during clear timer.

Test Plan: unit tests

Differential Revision: D55099773




cc @mrshenli @pritamdamania87 @zhaojuanmao @satgera @rohan-varma @gqchen @aazzolini @osalpekar @jiayisuse @H-Huang @kwen2501 @awgu @penguinwu @fegin @XilunWu @wanchaol @fduwjj @wz337 @tianyu-l @wconstab @yf225 @chauhang @dzhulgakov

d4l3k reviewed (approved):
LGTM -- need to update the existing test which is now failing

gag1jain committed:
[Watchdog Timer] Clear timer for already terminated process (#122324)

Summary:

handling cases where worker process is terminated w/o releasing the timer request, this scenario causes reaping of process at expiry.

removing the non-existent process during clear timer.

w/o the fix already terminated process are tried to reap and generate following messages and log watchdog event (timer expired), which are misleading:
```
Reaping worker_pid=[1966739]. Expired timers: ['test_scope']
Process with pid=1966739 does not exist. Skipping
Successfully reaped worker=[1966739] with signal=9
```

Test Plan: unit tests

Reviewed By: d4l3k

Differential Revision: D55099773
